### PR TITLE
feat:(555, 554, 550) Search Microservice

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves (555, 554, 550)

# Explain the feature
This PR enables the functionality of search for the One Platform. This Consist of Microservice integrated with the API Gateway and the SSI. Microservice is integrated with Apache Solr and provides functionality for data indexing.

## Does this PR introduce a breaking change
Yes

## Screenshots

<details>
<summary>View Screenshots</summary>

Test Status
image

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/search-service/package.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    "className": "Search",
+    "serviceName": "search"
+  },
+  "version": "0.0.1",
+  "keywords": [
+    "graphql",
+    "search",
+    "apache",
+    "solr",
+    "oneplarform"
```
Comment (deshmukhmayur) [inline packages/search-service/package.json:14]:
```suggestion
    "one-platform"
```

File: packages/search-service/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+**/bin
+**/charts
+**/docker-compose*
+**/Dockerfile*
+**/node_modules
+**/npm-debug.log
+**/obj
+**/secrets.dev.yaml
+**/values.dev.yaml
+README.md
```
Comment (deshmukhmayur) [inline packages/search-service/.dockerignore:24]:
```suggestion
README.md

```

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]
Some very minor typos.

**Review by deshmukhmayur — APPROVED** [review]

**Review by ghanlohar — APPROVED** [review]
Looks good.

**Pull request merged**